src/Tribe/Views/V2/Views/Widgets/Widget_View.php

```diff
@@ -61,6 +62,32 @@ protected function setup_template_vars() {
 		return $template_vars;
 	}
 
+	/**
+	 * Sets up the View repository arguments from the View context or a provided Context object.
+	 *
+	 * @since 4.9.3
+	 *
+	 * @param  Context|null $context A context to use to setup the args, or `null` to use the View Context.
+	 *
+	 * @return array The arguments, ready to be set on the View repository instance.
+	 */
+	protected function setup_repository_args( Context $context = null ) {
+		$context     = null !== $context ? $context : $this->context;
+		$context_arr = $context->to_array();
+		$args        = parent::setup_repository_args( $context );
+
+		/**
+		 * A widget-specific filter for repository args, based on widget slug.
+		 * Allows other plugins to add/remove args for the repository pre-query.
+		 *
+		 * @param array<string,mixed> $args        The arguments, ready to be set on the View repository instance.
+		 * @param array<string,mixed> $context_arr The context to use to setup the args,in array format.
+		 */
+		$args = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_repository_args", $args, $context_arr );
```

Comment:
```suggestion
		$args = apply_filters( "tribe_events_views_v2_{$this->get_slug()}_repository_args", $args, $context );
```

Just pass a reference to the context, dumping the context to array is inefficient and an eager resolution of an object optimized for JIT resolution.